src/main.rs

```diff
@@ -1,23 +1,115 @@
-use llvm_ir::{Function, Module};
+use std::{collections::HashMap, option::Option::Some};
+
+use llvm_ir::{BasicBlock, Module, Operand, constant::Constant, instruction::Instruction, name::Name::{Name, Number}, operand::Operand::{LocalOperand, ConstantOperand, MetadataOperand}, terminator::Terminator};
+
+struct BFInterp {
+    mem: HashMap<String, u64>,
+}
+
+impl BFInterp {
+
+    fn new() -> BFInterp {
+        BFInterp { 
+            mem: HashMap::new(),
+        }
+    }
+
+    fn interpret(&mut self, module: Module) -> Result<(), &'static str> {  
+        let main_func = match module.get_func_by_name("main") {
+            Some(main) => main,
+            None => return Err("No main function"),
+        };
+
+        match self.it_bb(&main_func.name, &main_func.basic_blocks) {
```

Comment:
Before I forget, something else I noticed: We already know that we can't just interpret all blocks in the order they appear. So let's just get rid of the for loop in `it_bb`, and have that function just interpret a single block, and just pass block 0 here.